icu4c/source/common/udata.cpp

```diff
@@ -751,12 +751,19 @@ openCommonData(const char *path,          /*  Path from OpenChoice?          */
 
     UDataPathIterator iter(u_getDataDirectory(), inBasename, path, ".dat", TRUE, pErrorCode);
 
+    UErrorCode mapFileStatus = U_ZERO_ERROR;
+
     while((UDataMemory_isLoaded(&tData)==FALSE) && (pathBuffer = iter.next(pErrorCode)) != NULL)
     {
 #ifdef UDATA_DEBUG
         fprintf(stderr, "ocd: trying path %s - ", pathBuffer);
 #endif
-        uprv_mapFile(&tData, pathBuffer);
+        uprv_mapFile(&tData, pathBuffer, &mapFileStatus);
```

Comment:
Could we pass pErrorCode directly to uprv_mapFile()?
Also, UDataPathIer.next(pErrorCode) can also potentially set a U_MEMORY_ALLOCATION_ERROR.
In the line below here, could probably bail out on any kind of failure.